core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -244,5 +242,11 @@ oppia.controller('CollectionPlayer', [
           'There was an error loading the collection.');
       }
     );
+
+    $scope.$watch('collection', function(newValue) {
```

Comment:
@kevintab95 When does the value of 'collection' change? (I was just wondering why we couldn't have a simple direct call to $scope.generatePathParameters() here.)